#### 概念

异常是程序在执行期间产生的问题。C++异常是指在程序运行时发生的特殊情况，比如尝试除以零的操作。

异常提供了一种转移程序控制权的方式。C++,异常处理涉及到三个关键字：**try、catch、throw。**

- throw: 当问题出现时，程序会抛出一个异常。这是通过使用 throw 关键字来完成的。

- catch: 在您想要处理问题的地方，通过异常处理程序捕获异常。catch 关键字用于捕获异常。

- try: try 块中的代码标识将被激活的特定异常。它后面通常跟着一个或多个 catch 块。


#### 抛出异常
```
double division(int a, int b)
{
   if( b == 0 )
   {
      throw "Division by zero condition!";
   }
   return (a/b);
}
```

#### 捕获异常

```
try
{
   // 保护代码
}catch( ExceptionName e )
{
  // 处理 ExceptionName 异常的代码
}
```

捕获任何异常
```
catch(...) 
```

#### 例子  
```
#include <iostream>
using namespace std;
 
double division(int a, int b)
{
   if( b == 0 )
   {
      throw "Division by zero condition!";
   }
   return (a/b);
}
 
int main ()
{
   int x = 50;
   int y = 0;
   double z = 0;
 
   try {
     z = division(x, y);
     cout << z << endl;
   }catch (const char* msg) {
     cerr << msg << endl;
   }
 
   return 0;
}
```

- 在throw处抛出异常后,正常如果没有try语句那么程序执行将失败,在opencv多个程序中有遇到.
- throw抛出的异常有catch捕捉,按照种类进行筛选执行.一般的字符串为 const char* 类型


#### C++ 标准异常

在C++ 中一件定义了多种标准异常,也可以自己定义异常.
